R/class-CorrelationResult.R

```diff
@@ -0,0 +1,34 @@
+check_correlation_result <- function(object) {
+    errors <- character()
+
+    if (!length(object@data1Metadata) > 0) {
+      msg <- "Compute result must include information about data1 in the data1Metadata slot"
+      errors <- c(errors, msg)
+    } else if (!length(object@data2Metadata) > 0) {
+      msg <- "Compute result must include information about data2 in the data2Metadata slot"
+      errors <- c(errors, msg) 
+    }
+
+    return(if (length(errors) == 0) TRUE else errors)
+}
+
+
+#' Corrleation Result
+#' 
+#' A class to represent the computed results from a correlation calculation.
+#' This includes their representation in R, as JSON, and how they are written to files.
+#' 
+#' @slot statistics An optional slot of any values. List or data.frame are recommended. It is not required to have rows or cols map to samples.
+#' @slot data1Metadata A character describing data1. Should at least describe the name of data1.
+#' @slot data2Metadata A character describing data2. Should at least describe the name of data2.
+#' @name CorrelationResult-class
+#' @rdname CorrelationResult-class
+#' @export
+CorrelationResult <- setClass("CorrelationResult", representation(
```

Comment:
im not quite happy w calling this a result, but also the ComputeResult class a result, and then assigning this CorrelationResult to the stats slot of ComputeResult. i like the overall strategy, but not the name of this particular class... except i dont have a good suggestion off the top of my head, outside of avoiding reusing the word result here. is CorrelationStatisticsAndMetadata too much?

Reply:
Luckly it doesn't have to be specific to the assay v metadata because of the dataXMetadata slots. But we could call it CorrelationOutput, or CorrelationData? Or another way to get at it, why is the correlation output different than the DifferentialAbundanceResult? Or are we not happy with either being called 'result'?

Reply:
i agree its not different to differential abundance, i just didnt notice it before. but both places we have a class called *Result that gets assigned to the 'statistics' slot of the ComputeResult.. and both cases are in fact statistics, and some associated metadata to help interpret those statistics. thats ideally what the name should capture.

Reply:
#58